File: problems/1768/1768.merge-strings-alternately.go
/*
 * @lc app=leetcode id=1768 lang=golang
 *
 * [1768] Merge Strings Alternately
 */

package solution

// @lc code=start
func mergeAlternately(word1 string, word2 string) string {
	res := ""
	i, j := 0, 0

	for i < len(word1) && j < len(word2) {
		res += string(word1[i]) + string(word2[j])
		i++
		j++
	}

	for i < len(word1) {
		res += string(word1[i])
		i++
	}

	for j < len(word2) {
		res += string(word2[j])
		j++
	}

	return res
}

// @lc code=end

File: problems/1768/1768.merge-strings-alternately.py
#
# @lc app=leetcode id=1768 lang=python3
#
# [1768] Merge Strings Alternately
#


# @lc code=start
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i, j = 0, 0
        res = ""
        while i < len(word1) and j < len(word2):
            res += word1[i] + word2[j]
            i += 1
            j += 1

        if i < len(word1):
            res += word1[i:]
        if j < len(word2):
            res += word2[j:]

        return res


# @lc code=end
